ament_tools/build_types/cmake.py

```diff
@@ -167,12 +177,15 @@ def _common_cmake_on_test(self, context, build_type):
         prefix = self._get_command_prefix('test', context)
         if not IS_WINDOWS:
             if has_make_target(context.build_space, 'test') or context.dry_run:
-                if MAKE_EXECUTABLE is None:
-                    raise VerbExecutionError("Could not find 'make' executable")
-                cmd = prefix + [MAKE_EXECUTABLE, 'test']
-                if 'ARGS' not in os.environ:
-                    cmd.append('ARGS="-V"')
-                yield BuildAction(cmd)
+                if CTEST_EXECUTABLE is None:
+                    raise VerbExecutionError("Could not find 'ctest' executable")
+                # invoke CTest directly in order to pass arguments
+                yield BuildAction(prefix + [
+                    CTEST_EXECUTABLE,
+                    # show all test output
+                    '-V',
+                    '--force-new-ctest-process'] +
+                    context.ctest_args)
```

Comment:
I don't think so. `ctest` does not use the environment variable `ARGS` at all. The value of it does not have any effect in the following command:

    ARGS="--help" ctest

The `test` target in a generated `Makefile` looks like this:

    test:
    	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running tests..."
    	/usr/bin/ctest --force-new-ctest-process $(ARGS)